File: Practica5/Java/PromedioYDesviacion/Estadistica.java
package Practica5.Java.PromedioYDesviacion;
public class Estadistica {
    public static double calcularPromedio(double[] n) {
        double s = 0;
        for (double m : n) {
            s += m;
        }
        return s / n.length;
    }
    public static double calcularDesviacion(double[] n, double p) {
        double s = 0;
        for (double m : n) {
            s += Math.pow(m - p, 2);
        }
        return Math.sqrt(s / (n.length - 1));
    }
}

File: Practica5/Python/PromedioYDesviacion/Estadistica.py
import math
class Estadistica:
    def __init__(self, n):
        self.n = n
    def calcular_promedio(self):
        return sum(self.n) / len(self.n)
    def calcular_desviacion(self):
        p = self.calcular_promedio()
        return math.sqrt(sum((x - p) ** 2 for x in self.n) / (len(self.n) - 1))
    def mostrar_resultados(self):
        p = self.calcular_promedio()
        d = self.calcular_desviacion()
        print(f"El promedio es {p:.2f}")
        print(f"La desviación estándar es {d:.5f}")